### PR TITLE
[#38] Feature / Refactor DistributedAtomSpace constructor

### LINKED ISSUE #38: Refactor DistributedAtomSpace constructor
We'll have two use cases defined by a given tag parameter defaulted to `client`.

`server`: it means we're instantiating a DAS in the Openfaas/AWS to serve our lambda functions. In this case, the actual type of the AtomDB must be figured out via extra parameters. Currently we have only "redis-mongo".

`client`: it means we're instantiating a DAS to make queries. The actual AtomDB in this case should be 'ram_only'. 

Anyway a default list of remote DAS should be set to empty. Remote DAS can be added using a new method `attach_remote(ip)`, which will establish the connection to Openfaas/AWS and let it ready to use. Queries (in both cases) should be kept as is for while, using AtomDB to answer them. In another issue we'll add the functionality of choosing whether/when to use the list of remote DAS.

### PR DESCRIPTION
**Pull request by marcocapozzoli**
Resolves #38

### CODE REVIEW

File: hyperon_das/api.py [hunk 1]
```diff
@@ -26,44 +31,78 @@
...
     def __init__(
-        self,
-        database: DatabaseType,
-        host: Optional[str] = None,
-        port: Optional[str] = None,
+        self, das_type: DasType = DasType.CLIENT.value, **kwargs
     ) -> None:
-        self._db_type = database
+        self._type = das_type
+        self.__remote_das = []
```
Comment (andre-senna) [inline hyperon_das/api.py:43]: Why using two "_" as prefix? Does it have a special meaning?

File: hyperon_das/api.py [hunk 2]
```diff
@@ -26,44 +31,78 @@
...
-            raise InitializeServerException(
-                message='You must send the host parameter',
-                details=f'To use server type Das you must send at least the host parameter',
-            )
-
-        self.db = database_factory(DatabaseFactory(self._db_type), host, port)
+        if self._type == DasType.CLIENT.value:
+            self.db = InMemoryDB()
+        elif self._type == DasType.SERVER.value:
+            # TODO: Implement some like that in the future
```
Comment (andre-senna) [inline hyperon_das/api.py:65]: Please remove this comment and add a card in the board instead. This should be the standard procedure for TODOs. In this project.

File: hyperon_das/api.py [hunk 3]
```diff
@@ -26,44 +31,78 @@
...
+    @property
+    def remote_das(self):
+        return self.__remote_das
+
+    @retry(attempts=5, timeout_seconds=120)
+    def _connect_server(self, host: str, port: Optional[str] = None):
+        port = port or config.get("DEFAULT_PORT_OPENFAAS", '8080')
+        openfaas_uri = f'http://{host}:{port}/function/atomdb'
+        aws_lambda_uri = f'http://{host}/prod/atomdb'
+        url = None
```
Comment (andre-senna) [inline hyperon_das/api.py:86]: Please log connection attempts here as INFO

**Review by andre-senna — APPROVED** [review]
Please make the suggested fixes before merging

File: hyperon_das/api.py (hunk 2 above)
Comment (marcocapozzoli) [reply hyperon_das/api.py:65]: Ok

File: hyperon_das/api.py (hunk 1 above)
Comment (marcocapozzoli) [reply hyperon_das/api.py:43]: Because I did want to put it as a private attribute

**Pull request merged**